susemanager-utils/supportutils-plugin-susemanager/susemanager-connection-check.pl

```diff
@@ -95,11 +96,50 @@ sub run_query {
     };
 }
 
+# reposync computes subprocess count with min(os.cpu_count() * 2, 32)
+my $cpucount = 0;
+open(FILE, "< /proc/cpuinfo") and do
+{
+    $cpucount = scalar (map /^processor/, <FILE>);
+    close FILE;
+};
+my $reposyncmax = min($cpucount * 2, 32);
+print "Reposync connections: $reposyncmax\n";
+
+sub parse_salt_worker_threads {
+    my @config_files = sort {$b cmp $a} glob("/etc/salt/master.d/*.conf");
+    push(@config_files, "/etc/salt/master");
+
+    my $worker_threads = 0;
+    CONFIG: foreach my $config (@config_files) {
+        open(FILE, $config) and do
+        {
+            while (<FILE>)
+            {
+                my $line = $_;
+                next if($line =~ /^\s*#/);
+                $worker_threads = $1 if ($line =~ /worker_threads\s*:\s*(\d+)/);
+            }
+            close FILE;
+            last CONFIG if $worker_threads > 0;
+        };
+    };
+    return $worker_threads;
+};
+my $worker_threads = parse_salt_worker_threads();
+print "Salt worker threads: $worker_threads\n";
+# add one for mgr_events.py and for uyuni roster module
+# custom engines are not counted here as we don't know if they use (and
+# potentially leak) DB connections
+my $saltmax = $worker_threads + 2;
+
 # java web ui, taskomatic uses c3p0 pooling
 # search used a fixed number of connections (10)
 # + every apache process can eat a connection
-# + buffer for local connections
-my $mindb = (2*$javamax) + $apachemax + 10 + 60;
+# + every salt mworker (pillar rendering) + mgr_events.py engine + uyuni roster module
```

Comment:
Apparently salt-api will also do pillar rendering, we need to take another at the formula

Reply:
Maybe consider this value?
```
# Maximum number of actions targetting Salt SSH minions executing at the same time
taskomatic.sshminion_action_executor.parallel_threads = 20
```
From rhn.conf (default from https://github.com/uyuni-project/uyuni/blob/master/java/conf/rhn_java.conf)